lib/swaps/Swaps.ts

```diff
@@ -233,10 +290,89 @@ class Swaps extends EventEmitter {
     };
 
+    // add the deal. Going forward we can "record" errors related to this deal.
     this.addDeal(deal);
 
+    // verify LND setup. Make sure we are connected to BTC and LTC and that
+    // the peer is also connected to these networks.
+    const err = this.verifyLNDSetup(deal, peer);
+    if (err) {
+      this.setDealState(deal, SwapDealState.Error, err);
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    const heldAmount = this.checkAndHoldAmount(deal);
+    assert(heldAmount === 0 || heldAmount === deal.takerAmount,
+        'Partial amount is not yet supported');
+    if (heldAmount === 0) {
+      this.setDealState(deal, SwapDealState.Error, 'Can not swap. No capacity is available');
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    let lndclient: LndClient;
+    switch (deal.takerCurrency) {
+      case 'BTC':
+        lndclient = this.lndBtcClient;
+        break;
+      case 'LTC':
+        lndclient = this.lndLtcClient;
+        break;
+      default:
+        this.setDealState(deal, SwapDealState.Error, 'Can not swap. Unsupported Taker currency.');
+        this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+        return;
+    }
+
+    let height: number;
+    try {
+      const req = new lndrpc.QueryRoutesRequest();
+      req.setAmt(requestBody.takerAmount);
+      req.setFinalCltvDelta(requestBody.takerCltvDelta);
+      req.setNumRoutes(1);
+      req.setPubKey(peer.getLndPubKey(requestBody.takerCurrency)!);
+      const routes = await lndclient.queryRoutes(req);
+      deal.makerToTakerRoutes = routes.getRoutesList();
+      this.logger.debug('got ' + deal.makerToTakerRoutes.length + ' routes to destination: ' + deal.makerToTakerRoutes);
+      if (deal.makerToTakerRoutes.length === 0) {
+        this.setDealState(deal, SwapDealState.Error, 'Can not swap. Zero routes to destination found.');
+        this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+        return;
+      }
+    } catch (err) {
+      this.setDealState(deal, SwapDealState.Error, 'Can not swap. No route to destination found: ' + err.message);
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    try {
+      const info = await lndclient.getInfo();
+      height = info.getBlockHeight();
+      this.logger.debug('got block height of ' + height);
+    } catch (err) {
+      this.setDealState(deal, SwapDealState.Error, 'Can not swap. Unable to fetch block height: ' + err.message);
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    const routeCltvDelta = deal.makerToTakerRoutes[0].getTotalTimeLock() - height;
+
+    // cltvDelta can't be zero for both the LtcClient and BtcClient (constractor)
```

Comment:
typo in "constractor" - and maybe I'd also expand a little to say "(checked in constructor)".

Reply:
done